shinysdr/plugins/wspr/demodulator.py

```diff
@@ -225,6 +243,15 @@ def filename(self, start_time):
         time_str = time.strftime(b'%y%m%d_%H%M.wav', time.gmtime(start_time))
         filename = b'%s_%s' % (self.context.get_absolute_frequency_cell().get(), time_str)
         return os.path.join(self.directory, filename)
+    
+    @exported_value(type=unicode, label='Status', changes='explicit')
+    def get_status(self):
+        if self.__decoder_active:
+            return 'Decoding…'
+        elif self.__frequency_subscription and not self.__invalidated_by_frequency_change:
+            return 'Receiving'
```

Comment:
A timer or progress bar is a better idea. It looked harder to implement, so I didn't.

Also, just decoding-or-not + timer wouldn't convey the “frequency changed recently, so I'm definitely not receiving anything" part which I think is useful information particularly to newcomers to WSPR.

Also, a good progress bar implementation would actually send over the time to the client and compute the 'progress' client side so we're not sending a stream of predictably-updating timestamps/progresses to the client. I now think that a progress bar is worthwhile, but a project for later.